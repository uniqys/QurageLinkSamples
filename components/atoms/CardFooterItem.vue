<template>
  <a href="#" class="card-footer-item" @click="$emit('click')">
    <slot />
  </a>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator'

@Component
export default class CardFooterItem extends Vue {}
</script>
