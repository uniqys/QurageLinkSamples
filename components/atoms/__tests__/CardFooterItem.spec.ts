import { Vue } from 'nuxt-property-decorator'
import { shallowMount, Wrapper } from '@vue/test-utils'
import CardFooterItem from '~/components/atoms/CardFooterItem.vue'

describe('CardFooterItem.vue', () => {
  const slotContent = 'test content'

  let wrapper: Wrapper<Vue>
  beforeEach(() => {
    wrapper = shallowMount(CardFooterItem, {
      slots: {
        default: slotContent
      }
    })
  })

  it('should to match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit event click when pushed link', () => {
    wrapper.find('a').trigger('click')
    expect(wrapper.emitted('click')).toBeTruthy()
  })
})
