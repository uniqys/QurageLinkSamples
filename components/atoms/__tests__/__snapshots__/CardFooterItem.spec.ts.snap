// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardFooterItem.vue should to match snapshot 1`] = `
<a
  class="card-footer-item"
  href="#"
>
  test content
</a>
`;
